### PR TITLE
Custom hook for data fetching

### PR DESCRIPTION
**Pull request by sourabhmodi20051**
Adds Custom hook for data fetching.
Updates Workspaces page to use useFetch hook.

### CODE REVIEW

File: client/src/containers/Workspaces/Workspaces.js [hunk 1]
```diff
@@ -1,69 +1,58 @@
...
+import { Link } from "react-router-dom";
 
-import Container from '../../components/Container/Container';
-import Columns from '../../components/Columns/Columns';
-import InputField from '../../components/InputField/InputField';
-import SidebarList from '../../components/SidebarList/SidebarList';
-import Spinner  from '../../components/Spinner/Spinner';
- 
+import Container from "../../components/Container/Container";
+import Columns from "../../components/Columns/Columns";
```
Comment (rakeshtembhurne) [inline client/src/containers/Workspaces/Workspaces.js]: These components now can be pulled from a single source.

File: client/src/hooks/useFetch.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from "react";
+import callApi from "../libs/axios";
+
+const useFetch = (method, url, data) => {
+  const [response, setResponse] = useState(null);
+  const [error, setError] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    console.log("Use Effect called");
```
Comment (nimish-gupta) [inline client/src/hooks/useFetch.js]: remove console.logs

File: client/src/containers/Workspaces/Workspaces.js [hunk 3]
```diff
@@ -1,69 +1,58 @@
...
-    setJoinedWS(dummyWorkspaces);
-    setLoading(false);
-  }, []);
+  const { response, error, isLoading } = useFetch("GET", "/workspace", {});
 
   const dummyUser = {
     _id: 101010,
-    name: 'John',
-  }
+    name: "John"
```
Comment (rakeshtembhurne) [inline client/src/containers/Workspaces/Workspaces.js]: Do we need dummy user here? We have backend implemented. You use real data.

**Comment by sourabhmodi20051** [comment]
Adds useFetch hook for API calls.

File: client/src/hooks/useFetch.js (hunk 2 above)
Comment (sourabhmodi20051) [reply client/src/hooks/useFetch.js]: Done

File: client/src/containers/Workspaces/Workspaces.js (hunk 3 above)
Comment (sourabhmodi20051) [reply client/src/containers/Workspaces/Workspaces.js]: Reverted this file

File: client/src/containers/Workspaces/Workspaces.js (hunk 1 above)
Comment (sourabhmodi20051) [reply client/src/containers/Workspaces/Workspaces.js]: reverted this file

**Pull request merged**